### PR TITLE
[Reviewer: Rob] Validate json files in the upload_json script, and log when they fail

### PR DESCRIPTION
**Pull request by johadalin**
This now handles an empty file, and we have better, more consistent error messages.

@steelyvoid , you wanted notifications about this work

### CODE REVIEW

**Comment by johadalin** [comment]
This supercedes https://github.com/Metaswitch/sprout/pull/1779

**Comment by johadalin** [comment]
Rob, passing to you for a review, as you made comments on the old PR. 
I've tested this briefly on the staging system, and confirmed that it works as expected with a correct enum.json, an empty file, and an invalid json file.

File: sprout-base.root/usr/share/clearwater/clearwater-config-manager/scripts/upload_json [hunk 1]
```diff
@@ -76,20 +76,35 @@ uploadsize=10000
...
 
+# Ensure that the file is either empty, or valid JSON
+if [[ ! -s $CONFPATH/$FILENAME ]]
+then
+  echo "The configuration file at $CONFPATH/$FILENAME is empty, uploading a blank file"
+else
+  python -m json.tool < $CONFPATH/$FILENAME > /dev/null 2>&1
+  rc=$?
+  if [ $rc != 0 ]; then
+    echo "The configuration file at $CONFPATH/$FILENAME contains invalid json"
```
Comment (rkd-msw) [inline sprout-base.root/usr/share/clearwater/clearwater-config-manager/scripts/upload_json]: JSON (it's an acronym)

**Comment by rkd-msw** [comment]
Looks OK.

**Comment by johadalin** [comment]
This resolves https://github.com/Metaswitch/clearwater-issues/issues/1403

**Pull request merged**